File: Serene3/Serene3.Web/Modules/ServerTypes/com/cetogeryForm.ts
import { ImageUploadEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface cetogeryForm {
    Name: StringEditor;
    Photos: ImageUploadEditor;
    Description: StringEditor;
}

export class cetogeryForm extends PrefixedContext {
    static readonly formKey = 'com.cetogery';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!cetogeryForm.init) {
            cetogeryForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;

            initFormType(cetogeryForm, [
                'Name', w0,
                'Photos', w1,
                'Description', w0
            ]);
        }
    }
}

File: Serene3/Serene3.Web/Modules/com/cetogery/cetogeryForm.cs
using Serenity.ComponentModel;

namespace Serene3.com.Forms;

[FormScript("com.cetogery")]
[BasedOnRow(typeof(cetogeryRow), CheckNames = true)]
public class cetogeryForm
{
    public string Name { get; set; }
    public string Photos { get; set; }
    public string Description { get; set; }
}